### PR TITLE
Feature/CVE 2024 50623

### PR DESCRIPTION
**Pull request by PiranhaSa**
The provided regular expression is used to extract version information for specific software products (VLTrader, Harmony, LexiCom) from a given text or response. Here's a detailed breakdown of the pattern:

### Explanation of the Regex
```
(?:VLTrader|Harmony|LexiCom)/
(
  (?:[0-4](?:\.\d+){0,3}) 
  | 
  5(?:\.[0-7](?:\.\d{1,2}){0,2})? 
  | 
  5\.8(?:\.0(?:\.(?:0|[1-9]|1[0-9]|2[0-1]))?)?
)
\s*\(
```
1 -Non-capturing group: (?:VLTrader|Harmony|LexiCom)

- The (?:...) syntax is a non-capturing group used here to match one of the product names (VLTrader, Harmony, LexiCom) without including the match in the captured output. This ensures we are only interested in capturing the version number, not the product name.

2 - Product delimiter: /

- Ensures that the version number follows the product name with a /.

3- Main capturing group:

- The (...) is a capturing group that isolates and retrieves the version number. The version format can be:
     - Versions starting with 0–4: (?:[0-4](?:\.\d+){0,3})
            - Matches version numbers like 1, 3.2, 4.1.2.3.
     - Versions starting with 5 up to 5.7: 5(?:\.[0-7](?:\.\d{1,2}){0,2})?
            - Matches versions like 5, 5.3, 5.7.2.12.
     - Version 5.8 with finer granularity: 5\.8(?:\.0(?:\.(?:0|[1-9]|1[0-9]|2[0-1]))?)?
            - Matches 5.8, 5.8.0, 5.8.0.21.
 - The | operator ensures all these subformats are considered.
4  - Optional whitespace and parentheses: \s*\(

- Matches any optional whitespace (\s*) and ensures the version number is followed by an opening parenthesis (().

### Purpose of ?:
The ?: inside the groups (e.g., (?:VLTrader|Harmony|LexiCom) and (?:[0-4](?:\.\d+){0,3})) creates non-capturing groups. This allows matching specific patterns without including them in the result. In this regex, ?: ensures that only the version number is captured and returned, avoiding unwanted matches for the product name or separators.

---------------------------

### Added Checks for Response Headers
In addition to checking the body of the HTTP response, checks were added to analyze the headers as well.

### CODE REVIEW

File: agent/exploits/cve_2024_50623.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+from agent import definitions
+from agent import exploits_registry
+from agent.exploits import webexploit
+
+VERSION_PATTERN = re.compile(
+    r"(?:VLTrader|Harmony|LexiCom)/((?:[0-4](?:\.\d+){0,3})|5(?:\.[0-7](?:\.\d{1,2}){0,2})?|5\.8(?:\.0(?:\.(?:0|[1-9]|1[0-9]|2[0-1]))?)?)\s*\("
+)
+MAX_VULNERABLE_VERSION = version.parse("5.8.0.21")
+MIN_VULNERABLE_VERSION = version.parse("0.0.0")  # No minimum version specified
```
Comment (oussamaessaji) [inline agent/exploits/cve_2024_50623.py:14]:
```suggestion
MIN_VULNERABLE_VERSION = version.parse("0.0.0")
```

Comment (PiranhaSa) [reply agent/exploits/cve_2024_50623.py:14]: sure

File: tests/exploits/cve_2024_50623_test.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
+"""Unit tests for Agent Asteroid: CVE-2024-50623"""
+
+"""Unit tests for Agent Asteroid: CVE-2024-50623"""
```
Comment (nmasdoufi-ol) [inline tests/exploits/cve_2024_50623_test.py:3]:
```suggestion
"""Unit tests for Agent Asteroid: CVE-2024-50623"""
```

File: agent/exploits/webexploit.py [hunk 3]
```diff
@@ -86,14 +90,24 @@ def check(self, target: definitions.Target) -> list[definitions.Vulnerability]:
             return vulnerabilities
 
         for pattern in self.match_pattern:
-            if pattern.search(resp.text) is not None:
+            if (
+                pattern.search(resp.text) is not None
+                or pattern.search(str(resp.headers)) is not None
+            ):
```
Comment (mohsinenar) [inline agent/exploits/webexploit.py:96]: This is duplicate move it to a private method.

**Review by mohsinenar — DISMISSED** [review]

File: agent/exploits/webexploit.py (hunk 3 above)
Comment (PiranhaSa) [reply agent/exploits/webexploit.py:96]: I've only changed to make the detection better , and having it there seems more clear .

File: agent/exploits/cve_2024_50623.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
+"""Agent Asteroid implementation for CVE-2024-50623"""
+
+import re
+from packaging import version
+
+from agent import definitions
+from agent import exploits_registry
+from agent.exploits import webexploit
```
Comment (elyousfi5) [inline agent/exploits/cve_2024_50623.py:8]:
```suggestion
import re

from packaging import version

from agent import definitions
from agent import exploits_registry
from agent.exploits import webexploit
```

**Review by nmasdoufi-ol — APPROVED** [review]

**Review by 3asm — APPROVED** [review]

**Pull request merged**